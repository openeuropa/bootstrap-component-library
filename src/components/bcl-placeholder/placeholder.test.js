import { renderTwigFileAsNode, getVariants } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-placeholder/data";

const template = "@oe-bcl/bcl-placeholder/placeholder.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);
const variants = getVariants(false, ["link"]);
const sizes = ["lg", "xs", "sm"];
const animations = ["glow", "wave"];

describe("OE - Placeholder", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  sizes.forEach((size) => {
    test(`${size} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, size: size })
      ).resolves.toMatchSnapshot();
    });
  });

  animations.forEach((animation) => {
    test(`${animation} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, animation: animation })
      ).resolves.toMatchSnapshot();
    });
  });

  variants.forEach((variant) => {
    test(`${variant} renders correctly`, () => {
      expect.assertions(1);

      return expect(
        render({ ...demoData, variant: variant })
      ).resolves.toMatchSnapshot();
    });
  });
});
