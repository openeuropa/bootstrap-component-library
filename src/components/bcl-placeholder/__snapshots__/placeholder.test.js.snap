// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Placeholder danger renders correctly 1`] = `
<jest>
  <span
    class="placeholder bg-danger"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder dark renders correctly 1`] = `
<jest>
  <span
    class="placeholder bg-dark"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder glow renders correctly 1`] = `
<jest>
  <p
    class="placeholder-glow"
  >
    <span
      class="placeholder bg-primary"
      style="width: 70%;"
    />
  </p>
</jest>
`;

exports[`OE - Placeholder info renders correctly 1`] = `
<jest>
  <span
    class="placeholder bg-info"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder lg renders correctly 1`] = `
<jest>
  <span
    class="placeholder placeholder-lg bg-primary"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder light renders correctly 1`] = `
<jest>
  <span
    class="placeholder bg-light"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder link renders correctly 1`] = `
<jest>
  <span
    class="placeholder bg-link"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder primary renders correctly 1`] = `
<jest>
  <span
    class="placeholder bg-primary"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder renders correctly 1`] = `
<jest>
  <span
    class="placeholder bg-primary"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder secondary renders correctly 1`] = `
<jest>
  <span
    class="placeholder bg-secondary"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder sm renders correctly 1`] = `
<jest>
  <span
    class="placeholder placeholder-sm bg-primary"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder success renders correctly 1`] = `
<jest>
  <span
    class="placeholder bg-success"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder warning renders correctly 1`] = `
<jest>
  <span
    class="placeholder bg-warning"
    style="width: 70%;"
  />
</jest>
`;

exports[`OE - Placeholder wave renders correctly 1`] = `
<jest>
  <p
    class="placeholder-wave"
  >
    <span
      class="placeholder bg-primary"
      style="width: 70%;"
    />
  </p>
</jest>
`;

exports[`OE - Placeholder xs renders correctly 1`] = `
<jest>
  <span
    class="placeholder placeholder-xs bg-primary"
    style="width: 70%;"
  />
</jest>
`;
